### PR TITLE
Refactor Weblab dialog to redux + separate components

### PR DESCRIPTION
**Pull request by maddiedierker**
This PR refactors how dialogs are rendered in Weblab and moves them all to React components.

Previously, there were two types of dialogs:
1. [A hard-coded error dialog in `<WebLabView/>`](https://github.com/code-dot-org/code-dot-org/blob/44a592e83766775f62132a40acf262333308ff96/apps/src/weblab/WebLabView.jsx#L206-L229), controlled by `shouldShowError` in redux state
Screen Shot 2021-06-29 at 9 45 33 AM

2. [`renderErrorModal` in CdoBramble.js](https://github.com/code-dot-org/code-dot-org/blob/9eed87094483406121abf427a5702bd66757c1c0/apps/src/weblab/CdoBramble.js#L642-L700) that rendered an unstyled/non-React dialog (and also wasn't always an error dialog)
Screen Shot 2021-03-31 at 7 44 07 AM

This change:
- Refactors redux state to have a `dialog` (instead of `shouldShowError`). These dialogs are now generated in WebLab.js, which also triggers the redux action to render the dialog.
- Moves all dialogs to React components to clarify what dialogs/states are possible.

**All dialogs (none are new, just documenting them here):**

Upload error (type 1 above):
Screen Shot 2021-06-28 at 1 50 27 PM

Bramble failed to load (type 2 above):
Screen Shot 2021-06-28 at 1 44 45 PM

Generic fatal error from Bramble (type 2 above):
Screen Shot 2021-06-28 at 1 41 13 PM

Project reset failed (type 2 above - project resets are triggered by the red "Reset Web Lab" button):
Screen Shot 2021-06-28 at 1 48 21 PM

Project reset succeeded (type 2 above):
Screen Shot 2021-06-28 at 1 52 48 PM

## Links

- [STAR-1508](https://codedotorg.atlassian.net/browse/STAR-1508)

## Testing story

Manually tested. Follow-up work required here.

## Follow-up work

1. [STAR-1558](https://codedotorg.atlassian.net/browse/STAR-1558) (add another dialog that notifies users when they've used a disallowed HTML element)
2. [STAR-1617](https://codedotorg.atlassian.net/browse/STAR-1617) - Add storybook coverage for new React components as test coverage (doing this after STAR-1558 because I think that ticket will require some additional refactoring to this code)
3. Update our [Weblab support article](https://support.code.org/hc/en-us/articles/360016804871) with new screenshots

### CODE REVIEW

File: apps/src/componentLibrary/StylizedBaseDialog.jsx [hunk 1]
```diff
@@ -8,7 +8,41 @@ import Button from '@cdo/apps/templates/Button';
...
+ * Includes a FooterButton component that appropriately styles buttons for the dialog.
  */
+
+export function FooterButton(props) {
+  const isConfirm = props.type === 'confirm';
+  const isCancel = props.type === 'cancel';
+  const color = props.color || (isConfirm && 'orange') || (isCancel && 'gray');
+
+  // TODO: We shouldn't need to override <Button/> styles -- they should likely be default.
+  // Tracked by https://codedotorg.atlassian.net/browse/STAR-1616.
```
Comment (epeach) [inline apps/src/componentLibrary/StylizedBaseDialog.jsx:21]: Love a tracked todo!

**Review by epeach — APPROVED** [review]
This looks great - love to see some outdated stuff move to fit the rest of our code.

File: apps/i18n/weblab/en_us.json [hunk 2]
```diff
@@ -4,8 +4,15 @@
...
+  "fatalError": "Fatal Error: {message}. If you're in Private Browsing mode, data can't be written.",
+  "loadFailure": "We're sorry, Web Lab failed to load for some reason.",
   "refreshPreview": "Refresh and Save",
+  "reloading": "Reloading...",
+  "reset": "Reset Web Lab",
+  "resetComplete": "Web Lab Reset Complete",
+  "resetFailure": "Failed to reset Web Lab: {message}.",
   "toggleInspectorOn": "Inspector: Off",
   "toggleInspectorOff": "Inspector: On",
+  "troubleshootingSupport": "Please see our support article [\"Troubleshooting Web Lab problems\"]({url}) for more information.",
```
Comment (epeach) [inline apps/i18n/weblab/en_us.json:16]: Should the p in problems be capitalized?

Comment (maddiedierker) [reply apps/i18n/weblab/en_us.json:16]: it matches the title of the Zendesk article, so i think it's appropriate to leave as-is

**Pull request merged**